editor/src/messages/tool/tool_messages/path_tool.rs

```diff
@@ -1029,6 +978,44 @@ impl Fsm for PathToolFsmState {
 				}
 
 				match self {
+					Self::Ready => {
+						//closest segment should be set when pointermoved and tabhi update segment bhi hona chahiye
+
+						//check for a close segment if it is there
```

Comment:
Please put this functionality in a function to keep the tool code at least slightly readable.